lib/matplotlib/ticker.py

```diff
@@ -377,12 +381,32 @@ class FuncFormatter(Formatter):
     """
     Use a user-defined function for formatting.
 
-    The function should take in two inputs (a tick value ``x`` and a
-    position ``pos``), and return a string containing the corresponding
-    tick label.
+    The function can take in at most two inputs (a required tick value ``x``
+     and an optional position ``pos``), and must return a string containing
+     the corresponding tick label.
     """
     def __init__(self, func):
-        self.func = func
+        if not isinstance(func, types.BuiltinFunctionType):
+            nargs = len(inspect.signature(func).parameters)
+        elif (func.__name__ == 'format'):
```

Comment:
`isinstance(getattr(func, "__self__", None), str) and getattr(func, "__name__", "") == "format"` just to be safe...

Reply:
Little bit unclear here. Is the benefit here that in the first case default will end up falling through to the warning and in the second nargs will end up 0 and trigger the TypeError and those might be clearer for users if slightly less accurate?

Reply:
sometimes func won't have a __name__ attribute, but will be otherwise fine, so you don't want to error in that case.  sometimes some other object will have a method called format, in which case you don't want to go to this branch.

Reply:
but I only end up in this branch if it's a builtin, and I don't think there us any other builtin named format.